src/azul/indexer/log_forwarding_service.py

```diff
@@ -0,0 +1,100 @@
+import csv
+import gzip
+from typing import (
+    Iterator,
+)
+
+from botocore.response import (
+    StreamingBody,
+)
+from furl import (
+    urlsplit,
+)
+
+from azul import (
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.types import (
+    MutableJSON,
+)
+
+
+class LogForwardingService:
+    """
+    Parse logs from an Application Load Balancer.
+    Adapted from https://github.com/rupertbg/aws-load-balancer-logs-to-cloudwatch
+    """
+
+    def read_logs(self, bucket: str, key: str) -> Iterator[MutableJSON]:
+        response = aws.s3.get_object(Bucket=bucket, Key=key)
+        body: StreamingBody = response['Body']
+        with gzip.open(body, mode='rt') as f:
+            for i, row in enumerate(csv.reader(f, delimiter=' ', quotechar='"')):
```

Comment:
`quotechar='"'` uses a different syntax for escaping double quotes in fields:

```python
list(csv.reader(StringIO('a "He said ""Hello""!" b'), delimiter=' ', quotechar='"'))
[['a', 'He said "Hello"!', 'b']]
```

It should be mentioned that that syntax can't occur in the logs and why.